.circleci/config.yml

```diff
@@ -127,57 +127,128 @@ jobs:
           name: bundle install gems
           command: |
             $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle install
+            bundle install --path vendor/bundle
 
       - save_cache:
           name: save gem cache
           paths:
             - ./vendor/bundle
           key: v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
 
+      - restore_cache:
+          name: restore cross-compiled code
+          keys:
+            - compiled-<< pipeline.git.revision >>
+
       - run:
-          name: build openssl
-          no_output_timeout: 30m
+          name: restore cross-compiled code
           command: |
             $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle exec rake ports:openssl
+
+            $rubyArchitecture = (ruby -e 'puts RUBY_PLATFORM').Trim()            
+            $source = "C:\home\circleci\project\tmp\$rubyArchitecture\stage\lib\tiny_tds"
+            $destination = ".\lib\tiny_tds"
+            Get-ChildItem $source -Recurse -Exclude "*.rb" | Copy-Item -Destination {Join-Path $destination $_.FullName.Substring($source.length)}
```

Comment:
1. I recommend using relative paths if possible
2. Generally this job step seems a bit strange, and that's related to my understanding of moving the ports code to a separate job. If it were to happen that CircleCI cleared the workflow cache after the ports job ran but before this job ran, wouldn't that mean that this job would fail? Is it guaranteed that won't happen? What is the recommended way to transfer files between jobs?

Reply:
> If it were to happen that CircleCI cleared the workflow cache after the ports job ran but before this job ran, wouldn't that mean that this job would fail? Is it guaranteed that won't happen? What is the recommended way to transfer files between jobs?

Good point. So the Windows job has a `requires` declaration on the precompile job. So in case that one fails, the Windows jobs wouldn't even start.

Regarding cache, I did some more research here and looks like [workspaces are the better data store](https://circleci.com/docs/workspaces/), as they are meant to pass build artifacts down in CI. Didn't know about that before, but it's implement now with 2a071ad0. This also gets rid of the absolute paths.